tox.ini

```diff
@@ -0,0 +1,15 @@
+[tox]
+skipsdist = True
+envlist = upgrade
+
+[testenv]
+setenv =
+    PYTHONPATH={toxinidir}:{toxinidir}/src
+    PYTHONBREAKPOINT=ipdb.set_trace
+
+[testenv:upgrade]
+deps =
+    juju
+    pytest
+    pytest-operator
+commands = pytest -v --tb native --show-capture=no --log-cli-level=INFO -s {posargs} {toxinidir}/tests/upgrade
```

Comment:
It seems like if we're going to use tox, we should really adopt it, and have github actions just run `tox`, and have this look something like:

    pytest -k 'not upgrade'
    pytest -k 'upgrade'
    pytest -k 'not upgrade'

That will ensure that all of the regular tests work both before and after an upgrade

Reply:
Does that mean you are okay with using tox for the tests that are executed in every PR/merge?

Reply:
Sure, sounds good to me. We'll probably want a tox env for each bundle size, and also have tox accept extra params that it just passes to pytest, to make it easy to do things like `tox -e full -k selenium`